neopia/neosoco.py

```diff
@@ -439,9 +445,8 @@ def convert_scale(self, port='in1', omin=0, omax=255, tmin=0, tmax=100):
         else:
             raise TypeError
 
-    def check_color(self, port='in1', color='White'):
-        c_value = 0
-        if isinstance(port, str):
+    def check_color(self, port='in1', color='white'):
+        if isinstance(port and color, str):

```

Comment:
It's better adding value error checking to check the value a user inputed is in or out in the [_COLOR dictionary]( https://github.com/roboticsware/pylib_neobot/blob/3b19ed500b8a5c554da34830a760b3764a753524/neopia/neosoco.py#L62).